### PR TITLE
[New] Find closest facility from point

### PR DESCRIPTION
**Pull request by CalebRas**
## Description

This PR implements `Find closest facility from point` in `Routing and Logistics` category.
URL to README: [README](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-FindClosestFacilityFromPoint/Shared/Samples/Find%20closest%20facility%20from%20point)

## Linked Issue(s)

- `swift/issues/4648`

## How To Test

- Tap "Solve Routes" to solve and display the route from each incident to the closest facility.
- Tap "Reset" to reset the routes.

## Screenshots

find-closest-facility-from-point

### CODE REVIEW

**Comment by CalebRas** [comment]
Removed @mhdostal since he is already a reviewer on a current samples PR.

File: Shared/Samples/Find closest facility from point/FindClosestFacilityFromPointView.swift [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+import ArcGIS
+import SwiftUI
+
+struct FindClosestFacilityFromPointView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// A Boolean value indicating whether a routing operation is in progress.
+    @State private var routing = false
```
Comment (des12437) [inline Shared/Samples/Find closest facility from point/FindClosestFacilityFromPointView.swift:23]:
```suggestion
    @State private var isRouting = false
```
This Boolean does not read as an assertion. Feel free to use a different name than suggested.

**Review by des12437 — DISMISSED** [review]

File: Shared/Samples/Find closest facility from point/FindClosestFacilityFromPointView.swift [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+                
+                // Get the route for the facility and incident.
+                let closestFacilityRoute = closestFacilityResult.route(
+                    toFacilityAtIndex: closestFacilityIndex,
+                    fromIncidentAtIndex: incidentIndex
+                )
+                
+                // Create a graphic using the route's geometry.
+                return Graphic(geometry: closestFacilityRoute?.routeGeometry, symbol: routeSymbol)
+            }
```
Comment (njarecha) [inline Shared/Samples/Find closest facility from point/FindClosestFacilityFromPointView.swift:185]: Good to move closestFacilityRoute as part of guard as no point in creating the `Graphic` if route geometry is not available.
```suggestion
            let routeGraphics = incidentsIndices.compactMap { incidentIndex -> Graphic? in
                // Get the index for the facility closest to the given incident and facility route.
                guard let closestFacilityIndex = closestFacilityResult.rankedIndexesOfFacilities(
                    forIncidentAtIndex: incidentIndex
                ).first,
                      let closestFacilityRoute = closestFacilityResult.route(
                        toFacilityAtIndex: closestFacilityIndex,
                        fromIncidentAtIndex: incidentIndex
                      ) else {
                    return nil
                }
                
                // Create a graphic using the route's geometry.
                return Graphic(geometry: closestFacilityRoute.routeGeometry, symbol: routeSymbol)
            }
```

**Review by njarecha — CHANGES_REQUESTED** [review]
Looks good, just a comment and a suggestion.

File: Shared/Samples/Find closest facility from point/FindClosestFacilityFromPointView.swift [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    @StateObject private var model = Model()
+    
+    /// A Boolean value indicating whether a routing operation is in progress.
+    @State private var isRouting = false
+    
+    /// A Boolean value indicating whether routing is currently disabled.
+    @State private var routingIsDisabled = true
+    
+    /// A Boolean value indicating whether the error alert is showing.
+    @State private var errorAlertIsShowing = false
```
[the remark this answers is not in the document]
Comment (CalebRas) [reply Shared/Samples/Find closest facility from point/FindClosestFacilityFromPointView.swift:29]: Based on the discussion in `swift/issues/4648`, it looks this is going to get removed in Sprint X, so I think it is good to leave as is for now.

**Review by njarecha — APPROVED** [review]

**Review by des12437 — APPROVED** [review]

**Pull request merged**